### PR TITLE
feat: Enable configuring rollout strategy for Refinery deployment

### PR DESCRIPTION
**Pull request by bixu**
## Which problem is this PR solving?

We've observed Refinery updates to be disruptive in some cases. A rolling update with conservative defaults should help give Refinery nodes more time to rebalance traffic across the cluster.

### CODE REVIEW

**Comment by bixu** [comment]
CI is failing in a fairly opaque way. I'm happy to contribute more changes if someone can shed some light on the test suite behavior.

**Comment by TylerHelmuth** [comment]
@bixu the CI tests failed installing the chart.  Have you tested this change locally? I believe the Deployment spec is incorrect: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

**Comment by bixu** [comment]
@TylerHelmuth, good catch on the spec nesting. Fixed that, and testing locally (added a Brewfile to speed up local test setup):

```
ct install --config ./ct.yaml

<snip>

------------------------------------------------------------------------------------------------------------------------
 ✔︎ honeycomb => (version: "1.8.2", path: "charts/honeycomb")
 ✔︎ network-agent => (version: "0.3.0", path: "charts/network-agent")
 ✔︎ refinery => (version: "2.7.0", path: "charts/refinery")
------------------------------------------------------------------------------------------------------------------------
All charts installed successfully
```

I'm still fuzzy about why CircleCI is failing.

**Comment by TylerHelmuth** [comment]
@bixu it looks like you've committed some extra files.  Also please handle the merge conflict?

**Review by TylerHelmuth — COMMENTED** [review]
@bixu this looks good, but to make it more configurable for other strategy types lets implement something like we did in the collector helm chart: 
- https://github.com/open-telemetry/opentelemetry-helm-charts/blob/75f4606c5ac0fa76cf2a856d2c83c4f1edae732d/charts/opentelemetry-collector/values.yaml#L497-L502
- https://github.com/open-telemetry/opentelemetry-helm-charts/blob/75f4606c5ac0fa76cf2a856d2c83c4f1edae732d/charts/opentelemetry-collector/templates/deployment.yaml#L24-L31

Also we'll stick with [chart-testing](https://github.com/helm/chart-testing) as our linter.

**Comment by bixu** [comment]
> @bixu this looks good, but to make it more configurable for other strategy types lets implement something like we did in the collector helm chart:

I was hoping to get some feedback on the interface -- thanks! Will pull in this prior pattern.


> Also we'll stick with [chart-testing](https://github.com/helm/chart-testing) as our linter.

Noted :)

File: charts/refinery/values.yaml [hunk 1]
```diff
@@ -289,6 +289,13 @@ autoscaling:
   annotations: {}
     # my-annotation: some value
 
+rollout:
+  # Disable with `rollingUpdate: {}` if using a different `strategy``:
+  rollingUpdate:
+    maxUnavailable: 1
+    maxSurge: 1
```
Comment (TylerHelmuth) [inline charts/refinery/values.yaml:296]: To make sure this change is backwards compatible, let's default the value to `{}`.  We can leave the 2 entries underneath commented out.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**